### PR TITLE
Dedupe logic for both server and client

### PR DESCRIPTION
**Pull request by vijaye-statsig**
Added a Dedupe test to verify the numbers are good when concurrently calling logEvent

### CODE REVIEW

File: dotnet-statsig-tests/Server/DedupeTest.cs [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+using WireMock;
+using WireMock.Settings;
+using Statsig;
+using Statsig.Server;
+using Newtonsoft.Json.Linq;
+using WireMock.ResponseProviders;
+
+namespace dotnet_statsig_tests
+{
+    [Collection("Statsig Singleton Tests")]
```
Comment (dan-loomb) [inline dotnet-statsig-tests/Server/DedupeTest.cs]: If you are you the driver directly, I don't think you need this decorator.

File: dotnet-statsig-tests/Server/DedupeTest.cs [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    public class DedupeTest : IAsyncLifetime, IResponseProvider
+    {
+        WireMockServer _server;
+        string _baseURL;
+        int _flushedEventCount = 0;
```
Comment (dan-loomb) [inline dotnet-statsig-tests/Server/DedupeTest.cs:20]: nit: why do some start with _?

File: dotnet-statsig/src/Statsig/Client/ClientDriver.cs [hunk 3]
```diff
@@ -30,7 +30,7 @@ public class ClientDriver : IDisposable
         internal readonly string _clientKey;
         bool _disposed;
         RequestDispatcher _requestDispatcher;
-        EventLogger _eventLogger;
+        internal EventLogger _eventLogger;
```
Comment (dan-loomb) [inline dotnet-statsig/src/Statsig/Client/ClientDriver.cs:33]: Do you need this?

**Review by dan-loomb — APPROVED** [review]
Lgtm

File: dotnet-statsig-tests/Server/DedupeTest.cs (hunk 2 above)
Comment (vijaye-statsig) [reply dotnet-statsig-tests/Server/DedupeTest.cs:20]: Private fields start with _

File: dotnet-statsig/src/Statsig/Client/ClientDriver.cs (hunk 3 above)
Comment (vijaye-statsig) [reply dotnet-statsig/src/Statsig/Client/ClientDriver.cs:33]: Yes, for tests.

File: dotnet-statsig-tests/Server/DedupeTest.cs (hunk 1 above)
Comment (vijaye-statsig) [reply dotnet-statsig-tests/Server/DedupeTest.cs]: Will remove.

**Comment by dan-loomb** [comment]
Closing in favor of the private PR

**Pull request closed without merging**